### PR TITLE
fix: promise wrapper race/all

### PR DESCRIPTION
**Pull request by patrickhousley**
### Overview

Updates the promise wrapper race and all methods to accept any object that implements the `Iterable` interface.

Also creates unit tests for this condition and the Promise wrapper in general, and converts `jspdf.browser.js` to Jest tests.

### Related Issue(s)

https://issues.newrelic.com/browse/NEWRELIC-8148
https://github.com/newrelic/newrelic-browser-agent/issues/472

### Testing

Added unit tests that check race and all methods work with standard arrays, a customer iterable and a native iterable Set.

### CODE REVIEW

File: src/common/wrap/wrap-promise.test.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  test.each([null, undefined])('should not try to iterate a non-iterable %s', async (input) => {
+    jest.spyOn(globalScope.Promise, 'resolve')
+
+    await expect(globalScope.Promise.all(input)).rejects.toThrow()
+    expect(globalScope.Promise.resolve).not.toHaveBeenCalled()
+  })
+})
+
+describe('race', () => {
+  test('should acceptable iterables', async () => {
```
Comment (bjfield) [inline src/common/wrap/wrap-promise.test.js:91]:
```suggestion
  test('should work with acceptable iterables', async () => {
```

File: src/common/wrap/wrap-promise.test.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  const promiseInstance = new globalScope.Promise(jest.fn())
+
+  expect(promiseInstance).toBeInstanceOf(Promise)
+  expect(promiseConstructorCalls.length).toEqual(1)
+  expect(globalScope.Promise.toString()).toMatch(/\[native code\]/)
+  expect(globalScope.Promise.name).toEqual('Promise')
+})
+
+describe('all', () => {
+  test('should acceptable iterables', async () => {
```
Comment (bjfield) [inline src/common/wrap/wrap-promise.test.js:66]:
```suggestion
  test('should work with acceptable iterables', async () => {
```

**Review by bjfield — APPROVED** [review]

File: src/common/wrap/wrap-promise.js [hunk 3]
```diff
@@ -69,9 +69,9 @@ export function wrapPromise (sharedEE) {
       const prevStaticFn = prevPromiseObj[method]
       WrappedPromise[method] = function (subPromises) { // use our own wrapped version of "Promise.all" and ".race" static fns
         let finalized = false
-        subPromises?.forEach(sub => {
-          // eslint-disable-next-line
-          this.resolve(sub).then(setNrId(method === 'all'), setNrId(false));
+
+        Array.from(subPromises || []).forEach(sub => {
```
Comment (cwli24) [inline src/common/wrap/wrap-promise.js:73]: Not critical, but why not use `for...of` that works with any `iterable` class?

Comment (patrickhousley) [reply src/common/wrap/wrap-promise.js:73]: `for..of` was tried and yea it works but its also a .21% gzip code size increase for the spa loader due to the need for additional error prevention logic.

**Review by metal-messiah — APPROVED** [review]

**Pull request merged**